src/polymer/polymer2-element-scanner.ts

```diff
@@ -65,161 +57,183 @@ export class Polymer2ElementScanner implements JavaScriptScanner {
   async scan(
       document: JavaScriptDocument,
       visit: (visitor: Visitor) => Promise<void>): Promise<ScannedElement[]> {
-    const visitor = new ElementVisitor(document);
-    await visit(visitor);
-    return visitor.getRegisteredElements();
-  }
-}
+    const classFinder = new ClassFinder();
+    const elementDefinitionFinder =
+        new CustomElementsDefineCallFinder(document);
+    // Find all classes and all calls to customElements.define()
+    await Promise.all([visit(classFinder), visit(elementDefinitionFinder)]);
+
+    const elementDefinitionsByClassName = new Map<string, ElementDefineCall>();
+    const elementDefinitionsByClassExpression =
+        new Map<estree.ClassExpression, ElementDefineCall>();
+
+    for (const defineCall of elementDefinitionFinder.calls) {
+      if (defineCall.clazz.type === 'MaybeChainedIdentifier') {
```

Comment:
can you add a comment for those who might be unfamiliar with MaybeChainedIdentifier